lib/util.c

```diff
@@ -265,6 +265,7 @@ msp_strerror_internal(int err)
             break;
         case MSP_ERR_UNKNOWN_TIME_NOT_SUPPORTED:
             ret = "Kept mutations must have known mutation times.";
+            break;
```

Comment:
Oooo, nice catch. Starting to think most tests for exceptions should check the message too!

Reply:
The compiler spotted it - I don't know how it slipped through to be honest. It must have happened at some point when I fixed a merge conflict, but I don't know how CI passed afterwards. :shrug:

Reply:
@jeromekelleher (do I need to ping for comments like these?)

The error was introduced by https://github.com/tskit-dev/msprime/commit/f18650fdc2931bc88eaad334a4188f077ae83298 
This commit (and the ones after) passed all checks, however our CI is on gcc 6 and  `-Wimplicit-fallthrough` was added in gcc 7.

Reply:
Aha! Thanks for the detective work. This is unlikely to skip through again, so let's not worry about it.